### PR TITLE
Use `$XDG_STATE_HOME` to store history.db

### LINKED ISSUE #243: Make McFly history.db path configurable
Hey @cantino would you accept a PR to make the ` Settings::storage_dir_path()` configurable with an ENV var?

### PR DESCRIPTION
**Pull request by utkarshgupta137**
Fixes #243.

### CODE REVIEW

File: src/settings.rs [hunk 1]
```diff
@@ -355,35 +356,51 @@ impl Settings {
         settings
     }
 
-    // Use ~/.mcfly only if it already exists, otherwise create 'mcfly' folder in XDG_CACHE_DIR
+    // Use or create the 'mcfly' folder in `$XDG_CACHE_DIR``
     pub fn mcfly_training_cache_path() -> PathBuf {
-        let cache_dir = Settings::mcfly_xdg_dir().cache_dir().to_path_buf();
-
-        Settings::mcfly_base_path(cache_dir).join(PathBuf::from("training-cache.v1.csv"))
+        Settings::xdg_base_strategy()
```
Comment (cantino) [inline src/settings.rs:361]: What is the difference between mcfly_xdg_dir and xdg_base_strategy?

File: src/settings.rs [hunk 2]
```diff
@@ -355,35 +356,51 @@ impl Settings {
         settings
     }
 
-    // Use ~/.mcfly only if it already exists, otherwise create 'mcfly' folder in XDG_CACHE_DIR
+    // Use or create the 'mcfly' folder in `$XDG_CACHE_DIR``
```
Comment (cantino) [inline src/settings.rs:359]: Two ``

File: src/settings.rs [hunk 3]
```diff
@@ -355,35 +356,51 @@ impl Settings {
...
+    // Use or migrate to the 'mcfly' folder in `$XDG_STATE_DIR`
     pub fn mcfly_db_path() -> PathBuf {
-        let data_dir = Settings::mcfly_xdg_dir().data_dir().to_path_buf();
-
-        Settings::mcfly_base_path(data_dir).join(PathBuf::from("history.db"))
-    }
-
-    fn mcfly_xdg_dir() -> ProjectDirs {
-        ProjectDirs::from("", "", "McFly").unwrap()
+        let path = Settings::xdg_base_strategy()
```
Comment (cantino) [inline src/settings.rs:368]: Is this backwards compatible?

File: src/settings.rs (hunk 1 above)
Comment (utkarshgupta137) [reply src/settings.rs:361]: I could reuse the name if you want.

File: src/settings.rs (hunk 2 above)
Comment (utkarshgupta137) [reply src/settings.rs]: Fixed.

File: src/settings.rs (hunk 3 above)
Comment (utkarshgupta137) [reply src/settings.rs]: Sorry, forgot to migrate from data_dir, that's done now.

File: src/settings.rs (hunk 1 above)
Comment (cantino) [reply src/settings.rs:361]: I just didn't understand what the difference is between them.

Comment (utkarshgupta137) [reply src/settings.rs:361]: `mcfly_xdg_dir` returned a `ProjectDirs` from the `directories` crate. `xdg_base_strategy` base directory returns an `Xdg` struct from `etcetera` crate which is the counterpart to it.

**Comment by utkarshgupta137** [comment]
@cantino ping

File: README.md [hunk 4]
```diff
@@ -341,7 +341,10 @@ Note that only single-character-prompts are allowed. setting `MCFLY_PROMPT` to `
 
 ### Database Location
 
-McFly stores its SQLite database in the standard location for the OS. On OS X, this is in `~/Library/Application Support/McFly` and on Linux it is in `$XDG_DATA_DIR/mcfly/history.db` (default would be `~/.local/share/mcfly/history.db`). For legacy support, if `~/.mcfly/` exists, it is used instead.
```
Comment (cantino) [inline README.md:344]: I think instead of migrating, it should keep using the old location if it exists.

**Comment by o6uoq** [comment]
This PR has been open since April 2023 - is there an update if this is still going to be baked into McFly?

**Comment by praveenperera** [comment]
Closed by: https://github.com/cantino/mcfly/pull/244

**Pull request closed without merging**